frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx

```diff
@@ -176,13 +176,14 @@ export default function QuickSelect<
         };
 
   const handleClose = (
-    _: React.ChangeEvent<Record<string, never>>,
+    e: React.ChangeEvent<Record<string, never>>,
     reason: AutocompleteCloseReason
   ) => {
     if (reason === "toggleInput") {
       return;
     }
-    setOpen(false);
+    const nativeEvent: any = e.nativeEvent;
+    if (nativeEvent.relatedTarget?.id !== `${dataTestId}-id`) setOpen(false);
```

Comment:
```suggestion
    const { nativeEvent } = e;
    if (
      nativeEvent instanceof FocusEvent &&
      nativeEvent.relatedTarget instanceof Element &&
      nativeEvent.relatedTarget?.id !== `${dataTestId}-id`
    )
      setOpen(false);

```

Try this out

Reply:
Unfortunately, this breaks the behavior when the target clicked is not an Element.

Reply:
Also breaks when you try closing with the ESC key since it's a KeyboardEvent.

Reply:
```
    if (
      nativeEvent instanceof FocusEvent &&
      nativeEvent.relatedTarget?.id !== `${dataTestId}-id` ||
      !(nativeEvent instanceof FocusEvent)
    )
      setOpen(false);
```
this works!